### PR TITLE
[Autoscaler][Docker] Make disable_shm_size_detection more usable

### LINKED ISSUE #15022: [Autoscaler][Docker] Disable SHM Detection is Redundant
### Describe your feature request
If I manually specify `--shm-size` in `run_options`, I **also need to specify** `disable_shm_size_detection: true` inside the `docker` config. This behavior is confusing and rather redundant. 

For feature requests or questions, post on our Discussion page instead: https://discuss.ray.io/

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* This should be set to `True` if a user wants to manually set shm-size!

## Related issue number

Closes #15022

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ijrsvt** [comment]
cc @jennakwon06

File: doc/source/cluster/config.rst [hunk 1]
```diff
@@ -658,6 +658,7 @@ If enabled, Ray will not try to use the NVIDIA Container Runtime if GPUs are pre
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 If enabled, Ray will not automatically specify the size ``/dev/shm`` for the started container and the runtime's default value (64MiB for Docker) will be used.
+This must be set to ``True`` if ``--shm-size=<>`` is manually added to ``run_options``.
```
Comment (rkooo567) [inline doc/source/cluster/config.rst:661]: Explain that this is automatically bypassed?

File: doc/source/cluster/config.rst [hunk 2]
```diff
@@ -658,6 +658,7 @@ If enabled, Ray will not try to use the NVIDIA Container Runtime if GPUs are pre
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 If enabled, Ray will not automatically specify the size ``/dev/shm`` for the started container and the runtime's default value (64MiB for Docker) will be used.
+If ``--shm-size=<>`` is manually added to ``run_options``, this is *automatically* set to ``True``, meaning that Ray will defer to the user-provided value.
```
Comment (AmeerHajAli) [inline doc/source/cluster/config.rst:661]:
```suggestion
If ``--shm-size=<>`` is manually added to ``run_options``, this is *automatically* set to ``True`` (unless the user explicitly sets it to ``False``), meaning that Ray will defer to the user-provided value.
```

**Review by AmeerHajAli — APPROVED** [review]
LGTM. Thanks for adding a test!
I think it would be helpful to manually test as well (use ray up with --shm-size and without and see how it works)

**Review by rkooo567 — APPROVED** [review]
Nice change!

**Pull request merged**